### PR TITLE
Drand upgrade

### PR DESCRIPTION
**Pull request by Kubuxu**
Signed-off-by: Jakub Sztandera <kubuxu@protocol.ai>

### CODE REVIEW

File: chain/beacon/beacon.go [hunk 1]
```diff
@@ -18,18 +18,54 @@ type Response struct {
...
-func ValidateBlockValues(b RandomBeacon, h *types.BlockHeader, prevEntry types.BeaconEntry) error {
-	maxRound := b.MaxBeaconRoundForEpoch(h.Height, prevEntry)
+func ValidateBlockValues(bSchedule Schedule, h *types.BlockHeader, parentEpoch abi.ChainEpoch,
+	prevEntry types.BeaconEntry) error {
+	{
+		parentBeacon := bSchedule.BeaconForEpoch(parentEpoch)
+		currBeacon := bSchedule.BeaconForEpoch(h.Height)
+		if parentBeacon != currBeacon {
+			if len(h.BeaconEntries) != 2 {
+				return xerrors.Errorf("expected two beacon entries at beacon fork, got %d", len(h.BeaconEntries))
```
Comment (nikkolasg) [inline chain/beacon/beacon.go:55]: Why should there be entries from the two beacons here ? What does it bring ?
What if drand is completely broken at some point and you need to change ?

File: node/modules/lp2p/pubsub.go [hunk 2]
```diff
@@ -74,7 +74,7 @@ func GossipSub(in GossipIn) (service *pubsub.PubSub, err error) {
 	}
 
 	isBootstrapNode := in.Cfg.Bootstrapper
-	drandTopic, err := getDrandTopic(in.Dr.ChainInfoJSON)
+	drandTopic, err := getDrandTopic(in.Dr[0].Config.ChainInfoJSON)
```
Comment (magik6k) [inline node/modules/lp2p/pubsub.go:77]: We should switch to the new topic a bit before the fork, right?

Comment (magik6k) [reply node/modules/lp2p/pubsub.go:77]: I guess this is for setting up topic params, in which case it should probably be safe to set those for all beacons first (assuming those are different per beacon)

**Review by magik6k — APPROVED** [review]

File: node/modules/lp2p/pubsub.go (hunk 2 above)
Comment (Kubuxu) [reply node/modules/lp2p/pubsub.go:77]: Yeah, we can repeat-set them.

**Review by whyrusleeping — APPROVED** [review]

File: build/params_testnet.go [hunk 3]
```diff
@@ -12,6 +12,10 @@ import (
 	"github.com/filecoin-project/specs-actors/actors/builtin/power"
 )
 
+var DrandSchedule = map[abi.ChainEpoch]DrandEnum{
+	0: DrandIncentinet,
```
Comment (whyrusleeping) [inline build/params_testnet.go:16]: `build.UpgradeSmokeHeight: DrandMainnet, `

**Review by magik6k — APPROVED** [review]

**Pull request merged**